Add protocol suffix template

tarnap opened the pull request:
Protocols are usually stored in a physical medium with a signature.
With these changes it is possible to add a template which is appended to the protocol, intended for the signatures.

Resolves https://github.com/4teamwork/gever/issues/103

Rotonen commented on opengever/meeting/locales/de/LC_MESSAGES/opengever.meeting.po:
```diff
@@ -722,17 +722,17 @@ msgstr "Sitzung durchführen"
 
+#. Default: "Ad hoc agenda item template"
+#: ./opengever/meeting/committee.py:106
+#: ./opengever/meeting/committeecontainer.py:81
```
Rather just omit the line change pointer comments from the tree? They're going to go away soon enough anyway.

tarnap replied:
As discussed: I'm keeping the numbers in a separate commit to drop them right before merging the pull request.
Whenever i run `bin/i18n-build` these line number changes are unstaged changes and they bug me, making a temporary commit simplifies my workflow.

jone replied:
@tarnap https://github.com/4teamwork/opengever.core/pull/3442 is merged, you should rebase this one.

Rotonen commented on opengever/meeting/locales/de/LC_MESSAGES/opengever.meeting.po:
```diff
@@ -17,12 +17,12 @@ msgstr ""
 "Domain: DOMAIN\n"
 
-#: ./opengever/meeting/command.py:540
+#: ./opengever/meeting/command.py:560
```
We're going to drop the line number pointers from how we use gettext. This change is safe to leave out of the tree.

tarnap commented on opengever/meeting/committeecontainer.py:
```diff
@@ -21,14 +21,49 @@ def excerpt_template_constraint(value):
     return True
 
 
+def protocol_template_constraint(value):
```
Good point :)

tarnap commented on opengever/meeting/committee.py:
```diff
@@ -61,6 +61,18 @@ class ICommittee(form.Schema):
+    )
+
+    protocol_suffix_template = RelationChoice(
+        title=_('Protocol suffix template'),
```
Would you change the existing `protocol_template`-field as well (while I'm at it)?

jone replied:
@tarnap probably not, because it already exists that way and we will eventually drop it because it is only used in the old version..

jone commented on opengever/meeting/command.py:
```diff
@@ -361,6 +360,12 @@ def generate_file_data(self):
+            target_path = join(tmpdir_path, 'protocol_suffix_template.dox')
+            with open(target_path, 'wb') as fio:
+                document = self.meeting.get_protocol_suffix_template()
+                fio.write(document.file.data)
```
The `protocol_suffix_template` is a sablon template, so you need to "render" it with `Sablon`, same as with `protocol_header_template`.
Something like:
```python
suffix_sablon = Sablon(self.meeting.get_protocol_suffix_template())
sablon.process(self.document_operations.get_meeting_data(self.meeting).as_json())
with .. as fio:
    fio.write(suffix_sablon.file_data)
```

jone commented on opengever/meeting/command.py:
```diff
@@ -335,8 +335,7 @@ def update_protocol_document(self):
     def generate_file_data(self):
-        template = self.document_operations.get_sablon_template(self.meeting)
-        sablon = Sablon(template)
+        sablon = Sablon(self.meeting.get_protocol_header_template())
```
I'd rename that, probably to `header_sablon`, because the suffix will also be sablon..

tarnap commented:
@jone: I removed the suffix from the fixture and changed the `command.py`. I also let the meeting model return "None" if there is no suffix template assigned.
The tests pass but I don't trust the whole thing.
Whe should probably add a test which calls "generate_file_data" when a suffix template is provided.

tarnap commented:
@jone: please re-review this pull request.

jone commented on opengever/meeting/committeecontainer.py:
```diff
@@ -21,21 +21,47 @@ def excerpt_template_constraint(value):
         source=sablon_template_source,
-        required=True,
+        required=False,
+        constraint=required_with_word_feature,
```
Shouldn't that be `required_without_word_feature` ?

jone commented on opengever/meeting/committeecontainer.py:
```diff
@@ -21,21 +21,47 @@ def excerpt_template_constraint(value):
+                default='Protocol header template'),
+        source=sablon_template_source,
+        required=False,
+        constraint=required_without_word_feature,
```
Shouldn't that be `required_with_word_feature`?

jone commented on opengever/meeting/committeecontainer.py:
```diff
@@ -21,21 +21,47 @@ def excerpt_template_constraint(value):
         source=sablon_template_source,
         required=False,
-        constraint=excerpt_template_constraint,
+        constraint=required_with_word_feature,
```
Shouldn't that be `required_without_word_feature` ?

tarnap commented:
@jone: You were right, I got the features mixed up. I added the tests you wished into the last commit of this pull request. Please re-re-review :)

jone commented on opengever/meeting/tests/test_committee.py:
```diff
@@ -63,6 +63,62 @@ def test_get_excerpt_template_falls_back_to_container(self):
+            self.sablon_template)
+
+        self.assertEqual(
+            self.sablon_template, self.committee.get_excerpt_template())
```
`s/get_excerpt_template/get_protocol_template` 😏

jone commented on opengever/meeting/tests/test_committee.py:
```diff
@@ -63,6 +63,62 @@ def test_get_excerpt_template_falls_back_to_container(self):
+            self.sablon_template)
+
+        self.assertEqual(
+            self.sablon_template, self.committee.get_excerpt_template())
```
`s/get_excerpt_template/get_protocol_header` 😏

jone commented on opengever/meeting/tests/test_committee.py:
```diff
@@ -63,6 +63,62 @@ def test_get_excerpt_template_falls_back_to_container(self):
+            self.sablon_template)
+
+        self.assertEqual(
+            self.sablon_template, self.committee.get_excerpt_template())
```
`s/get_excerpt_template/get_protocol_suffix` 😏

tarnap replied:
i schäme mi grad e chli

tarnap commented:
@jone: changes made, please re-re-re-re-re-re-re-re-re-review :)

jone commented:
@tarnap can you rebase onto master for resolving the conflicts?
You may need to pop the intermediate merge commit though..

tarnap commented:
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉
